core/src/typ.rs

```diff
@@ -1149,89 +1149,14 @@ impl Traverse<RichTerm> for Type {
-                    _ => write!(f, "{tail}"),
-                }
-            }
-            RecordRowsF::Empty => Ok(()),
-            RecordRowsF::TailVar(id) => write!(f, " ; {id}"),
-            RecordRowsF::TailDyn => write!(f, " ; Dyn"),
-        }
-    }
-}
-
-impl Display for EnumRows {
+impl Display for Type {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self.0 {
-            EnumRowsF::Extend { ref row, ref tail } => {
-                write!(f, "'{row}")?;
+        use crate::pretty::*;
 
-                match tail.0 {
-                    EnumRowsF::Extend { .. } => write!(f, ", {tail}"),
-                    _ => write!(f, "{tail}"),
-                }
-            }
-            EnumRowsF::Empty => Ok(()),
-            EnumRowsF::TailVar(id) => write!(f, " ; {id}"),
-        }
-    }
-}
+        let allocator = pretty::BoxAllocator;
 
-impl Display for Type {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match &self.typ {
-            TypeF::Dyn => write!(f, "Dyn"),
-            TypeF::Number => write!(f, "Number"),
-            TypeF::Bool => write!(f, "Bool"),
-            TypeF::String => write!(f, "String"),
-            TypeF::Array(ty) => {
-                write!(f, "Array ")?;
-
-                if ty.fmt_is_atom() {
-                    write!(f, "{ty}")
-                } else {
-                    write!(f, "({ty})")
-                }
-            }
-            TypeF::Symbol => write!(f, "Sym"),
-            TypeF::Flat(ref t) => write!(f, "{}", t.pretty_print_cap(32)),
-            TypeF::Var(var) => write!(f, "{var}"),
-            TypeF::Forall { var, ref body, .. } => {
-                let mut curr: &Type = body.as_ref();
-                write!(f, "forall {var}")?;
-                while let Type {
-                    typ: TypeF::Forall { var, ref body, .. },
-                    ..
-                } = curr
-                {
-                    write!(f, " {var}")?;
-                    curr = body;
-                }
-                write!(f, ". {curr}")
-            }
-            TypeF::Enum(row) => write!(f, "[| {row} |]"),
-            TypeF::Record(row) => write!(f, "{{ {row} }}"),
-            TypeF::Dict {
-                type_fields,
-                flavour: DictTypeFlavour::Type,
-            } => write!(f, "{{ _ : {type_fields} }}"),
-            TypeF::Dict {
-                type_fields,
-                flavour: DictTypeFlavour::Contract,
-            } => write!(f, "{{ _ | {type_fields} }}"),
-            TypeF::Arrow(dom, codom) => match dom.typ {
-                TypeF::Arrow(_, _) | TypeF::Forall { .. } => write!(f, "({dom}) -> {codom}"),
-                _ => write!(f, "{dom} -> {codom}"),
-            },
-            TypeF::Wildcard(_) => write!(f, "_"),
-        }
+        let doc: DocBuilder<_, ()> = self.clone().pretty(&allocator);
+        doc.render_fmt(80, f)
```

Comment:
We have this definition for `Display` elsewhere for other types IIRC. Is there a way to deduplicate or remove that boilerplate?

Reply:
Right, at least we could make a small helper in `core/src/pretty.rs`, I think.